workflows/differentialabundance.nf

```diff
@@ -22,16 +22,24 @@ if (params.study_type == 'affy_array'){
     } else {
         error("CEL files archive not specified!")
     }
-} else{
-
-    // If this is not an affy array, assume we're reading from a matrix
-
-    if (params.matrix) {
+  // If this is another array platform and user wish to read from SOFT files
+  // then a GSE study identifier must be provided
+} else if (params.study_type == 'non_affy_array' && params.querygse != "" && params.features_metadata_cols != ""){
+    if (params.querygse) {
+        ch_querygse = Channel.of([exp_meta, params.querygse, params.features_metadata_cols])
```

Comment:
You don't need to place the param in a channel- it should be used directly from params when needed.

Reply:
Makes sense. I've removed it as input and put directly as external parameter. Sorry, still getting my head around how Nextflow works :)